File: src/dataset/go/rna_transcription_Exercism_go_rootulp/openAI/ChatGPT4_rna_transcription_v2.go
package strand

func ToRNA(dna string) string {
	var rna []byte
	for i := 0; i < len(dna); i++ {
		switch dna[i] {
		case 'C':
			rna = append(rna, 'G')
		case 'G':
			rna = append(rna, 'C')
		case 'T':
			rna = append(rna, 'A')
		case 'A':
			rna = append(rna, 'U')
		}
	}
	return string(rna)
}

File: src/dataset/c/rna_transcription_Exercism_attilahorvath/openAI/ChatGPT4_rna_transcription_v2.c
#include "rna_transcription.h"
#include <stdlib.h>

char *to_rna(const char *dna) {
    if (!dna) return NULL;

    char *rna = strdup(dna);
    if (!rna) return NULL;

    for (char *p = rna; *p; ++p) {
        switch (*p) {
        case 'G':
            *p = 'C';
            break;
        case 'C':
            *p = 'G';
            break;
        case 'T':
            *p = 'A';
            break;
        case 'A':
            *p = 'U';
            break;
        default:
            free(rna);
            return NULL;
        }
    }

    return rna;
}